builder.py

```diff
@@ -26,37 +17,33 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-import importlib
+import distutils.errors
+import subprocess
 import sys
-from pathlib import Path
 
-from setuptools import setup
+from setuptools import build_meta as _orig
+from setuptools.build_meta import *  # noqa: F401,F403
 
-ROOT = Path(__file__).parent.resolve()
-sys.path.insert(0, ROOT.as_posix())
+WAYLAND_FFI_BUILD = "./libqtile/backend/wayland/cffi/build.py"
 
 
-def can_import(module):
-    try:
-        importlib.import_module(module)
-    except Exception:
-        return False
-    return True
+def wants_wayland(config_settings):
+    if config_settings:
+        for key in ["Backend", "backend"]:
+            if config_settings.get(key, "").lower() == "wayland":
+                return True
 
+    return False
 
-def get_cffi_modules():
-    # Check we have cffi around. If not, none of these will get built.
-    if not can_import("cffi.pkgconfig"):
-        print("CFFI package is missing")
-        return
 
-    cffi_modules = []
+def build_wheel(wheel_directory, config_settings=None, metadata_directory=None):
+    """If wayland backend is requested, build it!"""
+    if wants_wayland(config_settings):
+        try:
+            from libqtile.backend.wayland.cffi.build import ffi
 
-    return cffi_modules
+            ffi.compile(verbose=True)
+        except (distutils.errors.DistutilsError, subprocess.CalledProcessError) as e:
+            sys.exit(f"Wayland backend requested but backend could not be built: {e}")
```

Comment:
Alternatively, we could always try to build the backend but only stop qtile building if the user has explicitly asked for wayland and the backend build fails.